src/hotspot/share/gc/parallel/psAdaptiveSizePolicy.hpp

```diff
@@ -227,6 +231,14 @@ class PSAdaptiveSizePolicy : public AdaptiveSizePolicy {
                        double gc_minor_pause_goal_sec,
                        uint gc_time_ratio);
 
+  void minor_collection_begin();
+  void minor_collection_end(GCCause::Cause gc_cause);
+
+  static bool should_update_eden_stats(GCCause::Cause cause) {
+    return AdaptiveSizePolicy::should_update_eden_stats(cause) ||
+           (UseIOPrioritySizePolicy && GCCause::_gc_locker == cause);
```

Comment:
comment explaining why (cause == _gc_locker) is checked here?

Reply:
When running spark on TPC-DS, we observed IO occurs simultaneously with jni calls. So we intend to update eden size upon GC locker caused young gc.